File: python/1455.check-if-a-word-occurs-as-a-prefix-of-any-word-in-a-sentence.py
#
# @lc app=leetcode id=1455 lang=python3
#
# [1455] Check If a Word Occurs As a Prefix of Any Word in a Sentence
#

# @lc code=start
class Solution:
    def isPrefixOfWord(self, sentence: str, searchWord: str) -> int:
        return next((i + 1 for i, word in enumerate(sentence.split()) if word.startswith(searchWord)), -1)
# @lc code=end

File: scala/1455.check-if-a-word-occurs-as-a-prefix-of-any-word-in-a-sentence.scala
/*
 * @lc app=leetcode id=1455 lang=scala
 *
 * [1455] Check If a Word Occurs As a Prefix of Any Word in a Sentence
 */

// @lc code=start
object Solution {
  def isPrefixOfWord(sentence: String, searchWord: String): Int = {
    sentence.split(" ").zipWithIndex.find {
      case (word, _) => word.startsWith(searchWord)
    }.map {
      case (_, index) => index + 1
    }.getOrElse(-1)
  }
}
// @lc code=end

